01_intro-r.qmd

```diff
@@ -399,34 +441,46 @@ These two features have allowed R users to contribute millions of lines of code
 
 The predecessor of R, the S programming language, was designed to call FORTRAN subroutines. Accordingly, many R packages used to call compiled FORTRAN code. Now, many R packaged call compiled C++ code. This gives users the intuition of R syntax with better performance. [Here's a brief history of R and S](https://www.r-project.org/conferences/useR-2006/Slides/Chambers.pdf).
 
-## CRAN
+### CRAN
 
 Most R packages are stored on the Comprehensive R Archive Network ([CRAN](https://cran.r-project.org/)). Packages must pass a modest number of tests for stability and design to be added to CRAN. 
 
-## `install.packages()`
+### `install.packages()`
 
 Packages can be directly installed from CRAN using `install.packages()`. Simply include the name of the desired package in quotes as the only argument to the function. 
 
 Installation need only happen once per computer per package version. It is customary to never include `install.packages()` in a `.R` script.
 
-### `r kableExtra::text_spec("Exercise", color = "#1696d2")`
+::: callout
+#### [`r paste("Exercise", exercise_number)`]{style="color:#1696d2;"}
+
+```{r}
+#| echo: false
+
+exercise_number <- exercise_number + 1
+```
 
 For practice, let's install the `RXKCD` package so we can view some comics in R. 
 
 ```
 install.packages("RXKCD")
 ```
 
-$$\cdot\cdot\cdot$$
+:::
 
-## `library()`
+### `library()`
 
 After installation, packages need to be loaded once per R session using the `library()` function. While `install.packages()` expects a quoted package name, it is best practice to use unquoted names in `library()`. 
 
 It is a good idea to include `library()` statements at the top of scripts for each package used in the script. This way it is obvious at the top of the script which packages are necessary.
 
-### `r kableExtra::text_spec("Exercise", color = "#1696d2")`
+#### [`r paste("Exercise", exercise_number)`]{style="color:#1696d2;"}
 
+```{r}
+#| echo: false
+
+exercise_number <- exercise_number + 1
+```
 For practice let's load the `RXKCD` package. 
```

Comment:
None of the images in this section render. Maybe we should replace RXKCD with some different examples anyway.

Reply:
updated to palmerpenguins